### PR TITLE
test: signin

### PR DESCRIPTION
**Pull request by KubaBaniak**
## TESTS FOR [THIS PR](https://github.com/KubaBaniak/RecipeAppAuth/pull/4)

### CODE REVIEW

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -70,18 +70,16 @@ export class AuthService {
...
-    const userCredentials =
+    const userPassword =
       await this.userCredentialsRepository.getUserCredentialsByUserId(
         request.userId,
       );
 
-    const hashedPassword = userCredentials ? userCredentials.password : null;
-
-    if (!hashedPassword) {
+    if (!userPassword) {
```
Comment (rad20c) [inline src/auth/auth.service.ts:78]: https://github.com/KubaBaniak/RecipeAppAuth/pull/6/files#r1360312226

File: test/auth.e2e-spec.ts [hunk 2]
```diff
@@ -72,4 +75,42 @@ describe('AuthController (e2e)', () => {
...
+        .set('Accept', 'application/json')
+        .send(userCredentials)
+        .expect((response: request.Response) => {
+          expect(response.body.accessToken).toBeDefined();
+          expect(typeof response.body.accessToken).toBe('string');
+        })
+        .expect(HttpStatus.OK);
+    });
+
+    it(`should throw 401 - wrong password`, async () => {
```
Comment (rad20c) [inline test/auth.e2e-spec.ts:103]: I'd also add a test when user does not exist

**Review by rad20c — CHANGES_REQUESTED** [review]
check comments - at least `userCredentials` mock should be changed

File: src/auth/__mocks__/user-credentials.repository.mock.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  storeUserCredentials(
+    userId: number,
+    password: string,
+  ): Promise<UserCredentials> {
+    return Promise.resolve({ userId, password });
+  }
+
+  async getUserCredentialsByUserId(): Promise<string | null> {
+    return Promise.resolve(faker.internet.password());
+  }
```
[the remark this answers is not in the document]
Comment (KubaBaniak) [reply src/auth/__mocks__/user-credentials.repository.mock.ts:15]: My bad. I've made naming mistake. Both functions return same thing: `Promise<string | null>` which is password or null but name says `getUserCredentials...`

**Review by rad20c — APPROVED** [review]

**Pull request merged**